Meteor 2.5 integration

RvanB opened the pull request:
Any changes to client/lib/semantic-ui can be ignored (auto-generated).

cabeese committed:
Disable 'images' collection in Mongo

The image handling in mongo changed and is no longer supported with
the version we use with Meteor 2.5. For the time being, we'll disable
anything related to mongo image handling.

cabeese committed:
misc code cleanup and fixes in in lib/ and server/

Small changes to fix would-be errors in newer versions of Node/Meteor.

RvanB committed:
Router and browser history fixes

RvanB committed:
Disable image store

RvanB committed:
Update semantic-ui files

RvanB committed:
Package updates, fixes

RvanB committed:
Delete unneeded semantic-ui files

RvanB committed:
Import LinkButton in FAQ

RvanB committed:
Switch from mongo atmosphere package to npm

RvanB committed:
Change banner message

cabeese commented on client/components/admin/AdminGamestate/imports/GamestateControls.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
+//import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/admin/AdminPuzzles/imports/CheckinPacketEditor.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/admin/AdminSponsors/AdminSponsors.jsx:
```diff
@@ -1,6 +1,8 @@
 import { Meteor } from 'meteor/meteor';
 import { withTracker } from 'meteor/react-meteor-data';
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/app/Footer.jsx:
```diff
@@ -3,8 +3,8 @@ import React, { Component } from 'react';
 
-Footer = class Footer extends Component {
-  render() {
+export default Footer = class Footer extends Component {
```
not sure why we need to define a variable - can't we just do `export default class Footer extends Component {`?

RvanB replied:
The Footer variable was removed, the change is to "export default class..."!

cabeese commented on client/components/app/imports/RequireAuth.js:
```diff
@@ -0,0 +1,8 @@
+  render() {
+    return <Authed accessLevel={this.props.accessLevel}>{this.props.children}</Authed>;
+  }
+}
```
nit: missing newline at end of file

cabeese commented on client/components/game/imports/ActivePuzzle.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { PropTypes } from 'react';
+// import React, { PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/CheckInPacket.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/GamePuzzle.jsx:
```diff
@@ -1,4 +1,6 @@
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/GameCountdown.jsx:
```diff
@@ -1,4 +1,6 @@
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/GamePuzzles.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/GameStats.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/GameUI.jsx:
```diff
@@ -1,5 +1,7 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/GameTeamWrapper.jsx:
```diff
@@ -1,6 +1,8 @@
 import { Meteor } from 'meteor/meteor';
-import React, { Component, PropTypes } from 'react';
-import { Link } from 'react-router';
+// import React, { Component, PropTypes } from 'react';
```
nit: remove this line

cabeese commented on client/components/game/imports/InactivePuzzle.jsx:
```diff
@@ -1,4 +1,6 @@
-import React, { PropTypes } from 'react';
+// import React, { PropTypes } from 'react';
```
clearly there are a bunch of these. let's go clean them all up at once later

cabeese commented on client/components/topbar/TopBar.jsx:
```diff
@@ -116,7 +118,7 @@ const volunteerMenuItems = [
 ];
 
-TopBar = class TopBar extends Component {
+export default TopBar = class TopBar extends Component {
```
TopBar is overwritten later in the file when we wrap it with some logic based on the state of the user (eg loggedIn? isAdmin?). Are we confident that adding this `export default` here is correct, or should it be later? There should be precedent for this other places that we use withTracker()

cabeese commented on client/components/volunteer/GameProgress.jsx:
```diff
@@ -1,6 +1,8 @@
 import { Meteor } from 'meteor/meteor';
 import { withTracker } from 'meteor/react-meteor-data';
-import React, {PropTypes} from 'react';
+// import React, {PropTypes} from 'react';
```
remove this line

cabeese replied:
and all others like it

cabeese commented on client/routes/accounts-config.jsx:
```diff
@@ -1,7 +1,7 @@
 
 Accounts.onLogin(() => {
-  const path = browserHistory.getCurrentLocation().pathname;
+  const path = browserHistory.location.pathname;
```
is this the preferred way to replace getCurrentLocation()? I remember some discussion in their docs about being careful touching some  properties/attributes so just want to make sure this is kosher

RvanB replied:
I believe the browserHistory variable is no longer the same type of history that we had before, I read somewhere to use history.location.pathname. I'll double check on this though!

RvanB replied:
Yes, this seems like the proper way according to the docs:
https://github.com/remix-run/history/blob/8117ab21f5e339fabaa6ed1d80290fe3cee40c27/docs/api-reference.md#history.location
Thanks for checking!

cabeese commented on client/routes/routes.jsx:
```diff
@@ -1,73 +1,93 @@
+import { browserHistory } from '../history';
 
-    <Route path='/' component={App}>
+{/* Custom Router: https://stackoverflow.com/a/70000286 */}
```
I don't think the { } is necessary for this comment since it's outside of a JSX-styled area

cabeese commented on package.json:
```diff
@@ -19,40 +19,42 @@
-    "react-dom": "^15.1.0",
-    "react-dropzone": "^3.12.2",
+    "@babel/runtime": "^7.15.4",
+    "bcrypt": "^5.0.1",
```
we have had issues with upgrading bcrypt before. are we sure this one is okay?

cabeese replied:
**Would like to address this before merging**

Though I think if we're not getting errors, and passwords are working as expected, then this may not be an issue...

cabeese commented on server/main.js:
```diff
@@ -5,7 +5,14 @@ import bodyParser from 'body-parser';
 * Setup Winston Logger
 ************************************************/
-const logger = require('winston');
+const { createLogger, transports, format } = require('winston');
```
I think Richard said he wasn't sure if these changes were necessary. Have we tried running without, and/or confirming our logging still works how we expect?

cabeese replied:
**Would like to address this before merging!**

cabeese commented on client/components/public/imports/SponsorRow.jsx:
```diff
@@ -27,7 +27,7 @@ class SponsorRowInner extends Component {
 }
 
-SponsorRowInner.PropTypes = {
+SponsorRowInner.propTypes = {
```
This seems like the only place where `propTypes` has been changed to be lowercase. Why?

cabeese reviewed (commented):
A bunch of comments here, most of which I think we should come back and address later (after merging and deploying)

cabeese committed:
/tickets API: Update to use WebApp to handle POST request

The "picker" library we previously used has not been maintained and
does not appear to be compatible with Meteor 2.5. Instead, we use the
pre-bundled WebApp wrapper to set up the /api/tickets route.

RvanB committed:
Temporary sponsor image fix

cabeese commented on lib/collections/sponsors.js:
```diff
@@ -40,23 +39,21 @@ Meteor.methods({
   'sponsors.updateImage'(ids) {
     check(ids, {
       sponsorId: String,
-      imageId: String,
```
can you leave this commented out so we remember to add it back?

RvanB committed:
Remove commented imports

cabeese committed:
/tickets API: test using 'bodyParser.urlencoded'

cabeese committed:
package.json: specify node@16.x for heroku

cabeese committed:
processTransaction: fix winston transport message

cabeese committed:
Remove Picker and add version message on startup
